src/Specs/model.json

```diff
@@ -64,28 +68,66 @@
                         "type": "object",
                         "description": "Relationship configuration (can not be empty)",
                         "properties": {
+                            "model": {
+                                "type": "string",
+                                "description": "The related model name (can be different from the key name)"
+                            },
                             "foreignKey": {
                                 "type": "string",
                                 "description": "The foreign key field name"
+                            },
+                            "ownerKey": {
+                                "type": "string",
+                                "description": "The key in the related model that this relationship references"
                             }
                         },
-                        "required": ["foreignKey"]
+                        "required": ["model", "foreignKey", "ownerKey"]
```

Comment:
@rajeshn95 These are not required. Because if these keys are not present in json, engine will auto-generate them. These are for overring.

Reply:
@rjvim Ok

Reply:
@rjvim Made the changes, only "foreignKey" is required now, as we can't leave the object empty.